### PR TITLE
feat: [PL-25091]: api for supporting autocomplete for default variables

### PR DESCRIPTION
**Pull request by tiwaritejas**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33060)

### CODE REVIEW

**Comment by tiwaritejas** [comment]
trigger messagecheck

File: 120-ng-manager/src/main/java/io/harness/ng/core/variable/services/impl/VariableServiceImpl.java [hunk 1]
```diff
@@ -210,6 +212,27 @@ public boolean delete(
...
+                    Criteria.where(VariableKeys.projectIdentifier).is(projectIdentifier)));
+  }
+
+  @Override
+  public List<String> getExpressions(String accountIdentifier, String orgIdentifier, String projectIdentifier) {
+    assureThatTheProjectAndOrgExists(accountIdentifier, orgIdentifier, projectIdentifier);
+    Criteria criteria = getCriteriaForVariableExpressions(accountIdentifier, orgIdentifier, projectIdentifier);
+    return variableRepository.findAIdByParentScope(criteria)
+        .stream()
+        .map(entity -> variableMapper.writeDTO(entity).getExpression())
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/variable/services/impl/VariableServiceImpl.java:232]: why do we need to write it to DTO, when we can get the required info (scope, id) from Entity itself?

**Review by archit-harness — COMMENTED** [review]
> Reviewed 4 of 11 files at r1, 3 of 6 files at r3, 1 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33060)* status: 13 file(s) to review. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/variable/resources/VariableResource.java` line 235 at r4](https://reviewable.io/reviews/harness/harness-core/33060#-N1wtzIV9PUIQcF9PFQA:-N1wtzIW8LXXNga9VD5J:b-9tafnb) ([raw file](https://github.com/harness/harness-core/blob/c0ee6db62cb99771809420f4f133e6c5f058a72c/120-ng-manager/src/main/java/io/harness/ng/core/variable/resources/VariableResource.java#L235)):*
> ```Java
>         ApiResponse(responseCode = "default", description = "Returns the Variable expressions.")
>       })
>   public ResponseDTO<List<String>>
> ```

this api should be hidden @meenakshi-harness  as it should not be exposed to customers



_Code quote:_
```Java
public ResponseDTO<List<Strin
```

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 11 files at r1, 1 of 6 files at r3, 2 of 3 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33060)* status: 13 file(s) to review. 1 of 2 approvals obtained.

**Review by archit-harness — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33060)* status: 13 file(s) to review. 1 of 2 approvals obtained.

**Review by meenakshi-harness — COMMENTED** [review]
> Reviewed 5 of 11 files at r1, 2 of 6 files at r3, 1 of 3 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33060)* status: 4 file(s) to review. 1 of 2 approvals obtained.

**Review by meenakshi-harness — APPROVED** [review]
> Reviewed 1 of 6 files at r3, 1 of 3 files at r4.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33060)* status: 2 file(s) to review.

**Review by meenakshi-harness — DISMISSED** [review]
> Reviewed 1 of 6 files at r3, 1 of 3 files at r4.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33060)* status: :shipit: complete! ✓

**Review by meenakshi-harness — APPROVED** [review]
> Reviewed 3 of 3 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33060)* status: 3 file(s) to review. 1 of 2 approvals obtained.

**Review by nishantsaini20 — APPROVED** [review]
> Reviewed 1 of 11 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33060)* status: 3 file(s) to review.

**Review by nishantsaini20 — APPROVED** [review]
> Reviewed 3 of 11 files at r1, 4 of 6 files at r3, 2 of 3 files at r4, 1 of 1 files at r5, 3 of 3 files at r6.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33060)* status: :shipit: complete! ✓

**Pull request merged**